dashboard/common/db_util.py

```diff
@@ -85,32 +84,46 @@ def get_default_user_course_id(user_id):
     return course_id
 
 
-def get_user_courses_info(username: str) -> List[Dict[str, Union[str, int, List[str]]]]:
+def get_user_courses_info(username: str, course_id=None) -> List[Dict[str, Union[str, int, List[str]]]]:
+    """
+    Fetching the user courses enrollment info, for standalone it will return all the courses enrollment for LTI
+    single course enrollment info
+    http://za.github.io/2015/06/29/django-filter-query-exception/
+    :param course_id: canvas short course id
+    :param username: user sis_name
+    :return: [{`course_id`: 1233, `course_name`: 'COURSES WN 2020', `enrollment_types`: ['StudentEnrollment'] }]
+    """
     logger.info(get_user_courses_info.__name__)
-    user_courses_info: List[Dict[str, Union[str, int, List[str]]]] = []
-    with django.db.connection.cursor() as cursor:
-        cursor.execute('''
-            SELECT c.canvas_id, c.name, u.enrollment_type
-            FROM user u
-            JOIN course c
-                ON u.course_id=c.id
-            WHERE u.sis_name= %s;
-        ''', [username])
-        courses = cursor.fetchall()
-        if courses is not None:
-            course_enrollments: Dict[int, Dict[str, Union[int, str, List[str]]]] = {}
-            for course in courses:
-                course_id, course_name, enrollment_type = course
-                if course_id not in course_enrollments.keys():
-                    course_enrollments[course_id] = {
-                        'course_id': course_id,
-                        'course_name': course_name,
-                        'enrollment_types': []
-                    }
-                course_enrollments[course_id]['enrollment_types'].append(enrollment_type)
-            user_courses_info = list(course_enrollments.values())
-    logger.info(f'User {username} is enrolled in these courses: {user_courses_info}')
-    return user_courses_info
+    course_enrollments: Dict[int, Dict[str, Union[int, str, List[str]]]] = {}
+    if course_id:
+        user_enrollments = User.objects.filter(course_id=canvas_id_to_incremented_id(course_id),
+                                               sis_name=username)
+    else:
+        user_enrollments = User.objects.filter(sis_name=username)
+    if user_enrollments.count() == 0:
+        logger.warning(
+            f'Couldn\'t find user {username} enrollment info could be admin or enrollment data not pulled in yet')
+        return []
+
+    for enrollment in user_enrollments:
+        enroll_type = enrollment.enrollment_type
+        course = int(incremented_id_to_canvas_id(enrollment.course_id))
+        if course not in course_enrollments.keys():
+            course_enrollments[course] = {
+                'course_id': course,
+                'course_name': '',
```

Comment:
I think your other code below will set this up, so it could probably be deleted. Maybe the thinking was this is a fallback, but the way your code is written now, it would never return a dictionary with an empty string for `course_name`.